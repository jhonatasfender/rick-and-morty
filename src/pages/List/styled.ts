import styled, { css } from "styled-components";

export const MessageError = styled.div`
  ${({ theme }) => css`
    width: 100%;
    color: ${theme.colors.white};
    font-size: 10rem;
    text-align: center;
    padding: 2rem;
  `}
`
