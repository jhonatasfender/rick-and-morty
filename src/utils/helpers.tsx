import { ReactNode } from 'react';
import { QueryClient, QueryClientProvider } from 'react-query';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';

import { render, RenderResult } from '@testing-library/react'

import theme from '../styles/theme'

const queryClient = new QueryClient()

export const renderWithThemeAndRoute = (children: ReactNode): RenderResult =>
  render(
    <QueryClientProvider client={queryClient}>
      <MemoryRouter>
        <ThemeProvider theme={theme}>{children}</ThemeProvider>
      </MemoryRouter>
    </QueryClientProvider>,
  )
