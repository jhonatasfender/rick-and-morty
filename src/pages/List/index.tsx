import Card from '../../components/Card'
import CardGrid from '../../components/CardGrid'
import Form from '../../components/Form'
import Loading from '../../components/Loading'
import Pagination from '../../components/Pagination'
import useList from '../../hooks/use-list'
import * as S from './styled'

export default function List(): JSX.Element {
  const {
    register,
    gatewaySubmit,
    data,
    isLoading,
    renge,
    page,
    setPage,
    errors,
  } = useList()

  if (!data || isLoading) return <Loading />

  return (
    <div>
      <Form errors={errors} register={register} gatewaySubmit={gatewaySubmit} />

      {data.results && (
        <>
          <CardGrid>
            {data.results.map((result) => (
              <Card key={result.id} result={result} />
            ))}
          </CardGrid>
          <Pagination renge={renge} page={page} setPage={setPage} />
        </>
      )}

      {data.error && <S.MessageError>{data.error}</S.MessageError>}
    </div>
  )
}
