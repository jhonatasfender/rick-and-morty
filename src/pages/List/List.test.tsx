import { act, logRoles, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { renderWithThemeAndRoute } from '../../utils/helpers'
import List from './'

describe('<List />', () => {
  it('should display a message when there is no result in the search', async () => {
    const { container } = renderWithThemeAndRoute(<List />)

    await waitFor(() => {
      expect(screen.getByText('Rick Sanchez')).toBeInTheDocument()
    })

    userEvent.type(screen.getByRole('textbox'), 'sdfgsdfgsdfgs')

    userEvent.click(screen.getByRole('button'))

    await waitFor(() => {
      expect(screen.getByText('There is nothing here')).toBeInTheDocument()
    })

    expect(container).toMatchSnapshot()
  })

  it('should load the component with default values', async () => {
    const { container } = renderWithThemeAndRoute(<List />)

    await waitFor(() => {
      expect(screen.getByText('Rick Sanchez')).toBeInTheDocument()
    })

    expect(container).toMatchSnapshot()
  })

  it('should display required field message', async () => {
    renderWithThemeAndRoute(<List />)

    await waitFor(() => {
      expect(screen.getByText('Rick Sanchez')).toBeInTheDocument()
    })

    userEvent.click(screen.getByRole('button'))

    await waitFor(() => {
      expect(screen.getByText('Este campo é obrigatório')).toBeInTheDocument()
    })
  })

  it('should search what was typed', async () => {
    renderWithThemeAndRoute(<List />)

    await waitFor(() => {
      expect(screen.getByText('Rick Sanchez')).toBeInTheDocument()
    })

    userEvent.type(screen.getByRole('textbox'), 'rick')

    userEvent.click(screen.getByRole('button'))

    await waitFor(() => {
      expect(screen.queryByText('Rick Sanchez')).not.toBeInTheDocument()
    })

    await waitFor(() => {
      expect(screen.getByText('Rick Sanchez')).toBeInTheDocument()
    })

    expect(screen.getAllByRole('listitem')).toHaveLength(25)
  })

  it('should page correctly according to the triggers', async () => {
    renderWithThemeAndRoute(<List />)

    await waitFor(() => {
      expect(screen.getByText('Rick Sanchez')).toBeInTheDocument()
    })

    let pagination = screen.getByRole('list', {
      name: 'Pagination Navigation',
    })

    const [prev, , , , , , next] = Array.from(
      pagination.childNodes,
    ) as Element[]

    userEvent.click(prev)

    userEvent.click(next)

    await waitFor(() => {
      expect(
        Array.from(pagination.childNodes).map((el) => el.textContent),
      ).toEqual(['', '1', '2', '3', '4', '...', '42', ''])
    })

    const [, , , , , , clickFortyTwo] = Array.from(
      pagination.childNodes,
    ) as Element[]

    userEvent.click(clickFortyTwo)

    await waitFor(() => {
      expect(
        Array.from(pagination.childNodes).map((el) => el.textContent),
      ).toEqual(['', '1', '...', '40', '41', '42', ''])
    })

    const [, , points] = Array.from(pagination.childNodes) as Element[]

    userEvent.click(points)

    await waitFor(() => {
      expect(
        Array.from(pagination.childNodes).map((el) => el.textContent),
      ).toEqual(['', '1', '...', '40', '41', '42', ''])
    })

    const [, , , , , , clickNextAgain] = Array.from(
      pagination.childNodes,
    ) as Element[]

    userEvent.click(clickNextAgain)

    await waitFor(() => {
      expect(
        Array.from(pagination.childNodes).map((el) => el.textContent),
      ).toEqual(['', '1', '...', '40', '41', '42', ''])
    })

    const [clickPrevisionAgain] = Array.from(pagination.childNodes) as Element[]

    userEvent.click(clickPrevisionAgain)

    await waitFor(() => {
      expect(
        Array.from(pagination.childNodes).map((el) => el.textContent),
      ).toEqual(['', '1', '...', '39', '40', '41', '42', ''])
    })
  })
})
