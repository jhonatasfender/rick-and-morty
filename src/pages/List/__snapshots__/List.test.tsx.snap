// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should display a message when there is no result in the search 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1rem;
  padding: 2rem;
  height: 10.5rem;
}

.c2 {
  border: 0.1rem solid #F6313C;
  border-radius: 0.4rem;
  height: 4rem;
  min-width: 30rem;
  width: 100%;
  padding: 1rem;
  background-color: #380E33;
  color: #ffffff;
  font-size: 1.4rem;
}

.c2:focus {
  outline: none;
}

.c3 {
  background-color: rgb(157 23 30);
  border: none;
  border-radius: 0.4rem;
  padding: 1rem 3rem;
  color: #ffffff;
  font-size: 1.4rem;
  font-weight: 600;
  cursor: pointer;
  height: 4rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c4 {
  width: 100%;
  color: #ffffff;
  font-size: 10rem;
  text-align: center;
  padding: 2rem;
}

<div>
  <div>
    <form
      class="c0"
    >
      <div
        class="c1"
      >
        <input
          class="c2"
          name="name"
          type="text"
        />
      </div>
      <button
        class="c3"
        type="submit"
      >
        Pesquisar
      </button>
    </form>
    <div
      class="c4"
    >
      There is nothing here
    </div>
  </div>
</div>
`;

exports[`<List /> should load the component with default values 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  overflow: hidden;
  border-radius: 0.8rem;
  width: 16rem;
  height: 30rem;
  border: 0.1rem solid #ffffff;
  cursor: pointer;
}

.c6 {
  min-width: 16rem;
  width: 16rem;
}

.c8 {
  color: #ffffff;
  font-size: 1.9rem;
  display: block;
  width: 13rem;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c7 {
  padding: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 1rem;
}

.c10 {
  font-size: 1rem;
  font-weight: bold;
  background-color: #947F18;
  color: #ffffff;
  padding: 0.4rem 1rem;
  border-radius: 1rem;
  min-width: 45%;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1rem;
  padding: 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1rem;
  padding: 2rem;
  height: 10.5rem;
}

.c2 {
  border: 0.1rem solid #F6313C;
  border-radius: 0.4rem;
  height: 4rem;
  min-width: 30rem;
  width: 100%;
  padding: 1rem;
  background-color: #380E33;
  color: #ffffff;
  font-size: 1.4rem;
}

.c2:focus {
  outline: none;
}

.c3 {
  background-color: rgb(157 23 30);
  border: none;
  border-radius: 0.4rem;
  padding: 1rem 3rem;
  color: #ffffff;
  font-size: 1.4rem;
  font-weight: 600;
  cursor: pointer;
  height: 4rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
}

.c12 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c11 {
  color: #ffffff;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 10rem;
  gap: 1rem;
  width: 100%;
  list-style-type: none;
}

.c14 {
  --size: 5rem;
  background-color: rgb(157 23 30);
  font-size: 1.8rem;
  font-weight: bold;
  border-radius: 50%;
  width: var(--size);
  height: var(--size);
  text-align: center;
  line-height: var(--size);
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c15 {
  --size: 4rem;
  background-color: rgb(157 23 30);
  font-size: 1.8rem;
  font-weight: bold;
  border-radius: 50%;
  width: var(--size);
  height: var(--size);
  text-align: center;
  line-height: var(--size);
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c13 {
  width: 8rem;
  fill: rgb(157 23 30);
  cursor: pointer;
}

.c16 {
  width: 8rem;
  fill: rgb(157 23 30);
  cursor: pointer;
}

<div>
  <div>
    <form
      class="c0"
    >
      <div
        class="c1"
      >
        <input
          class="c2"
          name="name"
          type="text"
        />
      </div>
      <button
        class="c3"
        type="submit"
      >
        Pesquisar
      </button>
    </form>
    <div
      class="c4"
    >
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Rick Sanchez
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              Earth (C-137)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Morty Smith
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Summer Smith
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Female
            </div>
            <div
              class="c10"
            >
              Earth (Replacement Dimension)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Beth Smith
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Female
            </div>
            <div
              class="c10"
            >
              Earth (Replacement Dimension)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Jerry Smith
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              Earth (Replacement Dimension)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Abadango Cluster Princess
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Alien
            </div>
            <div
              class="c10"
            >
              Female
            </div>
            <div
              class="c10"
            >
              Abadango
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Abradolf Lincler
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              Earth (Replacement Dimension)
            </div>
            <div
              class="c10"
            >
              Genetic experiment
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Adjudicator Rick
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Agency Director
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              Earth (Replacement Dimension)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Alan Rails
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
            <div
              class="c10"
            >
              Superhuman (Ghost trains summoner)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Albert Einstein
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              Earth (C-137)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Alexander
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              Earth (C-137)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Alien Googah
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Alien
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Alien Morty
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Alien
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Alien Rick
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Alien
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Amish Cyborg
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Alien
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
            <div
              class="c10"
            >
              Parasite
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Annie
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Female
            </div>
            <div
              class="c10"
            >
              Earth (C-137)
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Antenna Morty
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
            <div
              class="c10"
            >
              Human with antennae
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Antenna Rick
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
            <div
              class="c10"
            >
              Human with antennae
            </div>
          </div>
        </div>
      </div>
      <div
        class="c5"
        role="listitem"
      >
        <img
          class="c6"
          src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
        />
        <div
          class="c7"
        >
          <span
            class="c8"
          >
            Ants in my Eyes Johnson
          </span>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              Human
            </div>
            <div
              class="c10"
            >
              Male
            </div>
            <div
              class="c10"
            >
              unknown
            </div>
            <div
              class="c10"
            >
              Human with ants in his eyes
            </div>
          </div>
        </div>
      </div>
    </div>
    <ul
      aria-label="Pagination Navigation"
      class="c11"
    >
      <svg
        aria-hidden="false"
        class="c12 c13"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M14.91 6.71a.996.996 0 0 0-1.41 0L8.91 11.3a.996.996 0 0 0 0 1.41l4.59 4.59a.996.996 0 1 0 1.41-1.41L11.03 12l3.88-3.88c.38-.39.38-1.03 0-1.41z"
        />
      </svg>
      <li
        class="c14"
      >
        1
      </li>
      <li
        class="c15"
      >
        2
      </li>
      <li
        class="c15"
      >
        3
      </li>
      <li
        class="c15"
      >
        ...
      </li>
      <li
        class="c15"
      >
        42
      </li>
      <svg
        aria-hidden="false"
        class="c12 c16"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9.31 6.71a.996.996 0 0 0 0 1.41L13.19 12l-3.88 3.88a.996.996 0 1 0 1.41 1.41l4.59-4.59a.996.996 0 0 0 0-1.41L10.72 6.7c-.38-.38-1.02-.38-1.41.01z"
        />
      </svg>
    </ul>
  </div>
</div>
`;
